### ISSUE TITLE
Reference not found

### ISSUE THREAD

**Issue by tkoomzaaskz** [pateketrueke/json-schema-faker#25]
I was encountering a problem with the example snippet from the readme. I just copied the code as-is:

```js
var jsf = require('json-schema-faker');

var schema = {
  type: 'object',
  properties: {
    user: {
      type: 'object',
      properties: {
        id: {
          $ref: '#/definitions/positiveInt'
        },
        name: {
          type: 'string',
          faker: 'name.findName'
        },
        email: {
          type: 'string',
          format: 'email',
          faker: 'internet.email'
        }
      },
      required: ['id', 'name', 'email']
    }
  },
  required: ['user'],
  definitions: {
    positiveInt: {
      type: 'integer',
      minimum: 0,
      minimumExclusive: true
    }
  }
};

var sample = jsf(schema);

console.log(sample.user.name);
```

saved it in `index.js` (or whatever file), ran `npm install json-schema-faker` and ran this node code: `node index.js`. And I got an error:

```
c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\lib\index.js:23
      throw e;
            ^
Error: Reference not found: http://json-schema.org/schema#/definitions/positiveInt
    at module.exports (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\find-reference.js:38:11)
    at clone (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:15:17)
    at clone (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:34:19)
    at clone (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:34:19)
    at clone (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:34:19)
    at clone (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:34:19)
    at Object.module.exports [as resolveSchema] (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\util\resolve-schema.js:55:10)
    at $ref (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\node_modules\deref\lib\index.js:48:14)
    at generate (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\node_modules\json-schema-faker\lib\index.js:18:21)

    at Object.<anonymous> (c:\Users\tomasz.ducin\Tests\jsf-ref-broken\index.js:35:14)
```

To run any `jsf` code, I had to *cut & paste* the reference to the schema inside (removing `definitions` key). Then it started working. So, more or less, it seems that the main example in the docs might not work as expected. Are there any tests for that? Or maybe it's platform-dependent - I'm working on windows, unfortunately.

**Comment by pateketrueke** [comment]
I've replicated the same issue. If you set an id for the schema it will work but it should be fine without the id.

**Comment by pateketrueke** [comment]
Solved in `deref@0.3.0` :beers:

https://github.com/gextech/deref#schema-identity

Also the specs was introducing an unexpected behavior by using `deref` for normalization: epic fail.
